lib/view_model/active_record/association_manipulation.rb

```diff
@@ -200,29 +201,37 @@ def delete_associated(association_name, associated_id, type: nil, deserialize_co
       end
 
       child_context = deserialize_context.for_child(self, association_name: association_name)
-      vm = direct_viewmodel.new(models.first)
-      child_context.visible!(vm)
-      initial_editability = child_context.initial_editability(vm)
-      child_context.editable!(vm,
-                              initial_editability: initial_editability,
-                              changes: ViewModel::Changes.new(deleted: true))
-      association.delete(vm.model)
-      vm
+      child_vm = direct_viewmodel.new(models.first)
+
+      child_context.run_callback(ViewModel::Callbacks::Hook::BeforeVisit, child_vm)
```

Comment:
This is quite a mouthful. We don't have any overlapping regions, so we could make some helpers that help balancing callbacks:

```end
def with_foo_callback(context, vm)
  context.run_callback(BeforeFoo, vm)
  yield
  context.run_callback(AfterFoo, vm)
end
```

Trading explicit calls for some indent levels

Reply:
I think I prefer the explicit calls, given how big the body in the yield is. What do you think?

Reply:
Either seems fine.

Reply:
Tried out the wrappers, on balance I think they're worth it.